### DISCUSSION TITLE
nice to have note conversion feature

### DISCUSSION THREAD

**Discussion by sugizo** [mido/mido#541] [Ideas]
nice to have note conversion feature

e.g.
note_scientific = C4
note_helmholtz = c'
note = C
octave = 4
frequency_integer = 262
frequency_float = 261.6
midi = 60

**ref**
https://sugizo.pythonanywhere.com/music/master/pitch (for data gathered from wiki)
https://en.wikipedia.org/wiki/Musical_note (for explaination)

if you interest, can share the API data for note above from the web (sugizo.pythonanywhere.com/music)

best regards

**Comment by rdoursenaud** [comment]
Thanks for the suggestion!

It's something I plan to add for mido 2.0.
See: https://github.com/mido/mido/wiki/Version-2-Design

I was thinking about a flexible `Note` object that could be passed to construct `Message` objects that require it and conversely, also decode note information from these.
Also allow using conventional notations to construct our objects instead of the MIDI note number.

I prefer a local API but I'd love to see what you've done.

**Reply by sugizo** [reply]
the idea is base on pretty_midi that already have this feature, 
but 
it's not complete (dont have helmholtz, frequency, etc), 

realize, many people use scientific note (i.e. C4), 
but 
the helmholtz still used till now

and perhaps some data like frequency is also needed by people

you can download the data from the web directly 
https://sugizo.pythonanywhere.com/music/master/pitch

available in json, csv, etc
but 
will attached here as promise on 1st posted (realized that can attached *.json, so here the *.csv)

[note_data.csv](https://github.com/mido/mido/files/12444021/note_data.csv)

**Reply by rdoursenaud** [reply]
Thanks, I'll check it out as soon as I start working on that feature.

**Comment by sugizo** [comment]
comparable to pretty_midi, midiutil, miditoolkit, music21

mido is the one that hard to learn and implement 
to 
generate and read midi 
for me

my problem is **HARD** to understand about delta time, 
while 
in other python modules mention above is easy, 
because 
either other modules are straight forward to define start time and end time
or
they have a good documentation and given the clear example

hopefully in mido 2.0, this will cover problem mention above 

thanks
